.github/workflows/nix.yaml

```
@@ -51,16 +51,20 @@ jobs:
           - system: aarch64-linux
             host: aarch64-darwin
-    # needs: [process-labels]
+            run-condition: ${{ github.ref == 'refs/heads/main' }}
+    needs: [process-labels]
     runs-on: ${{ matrix.host }}
-    # needs.process-labels.outputs.has-backend == 'true' && 
-    #if: |
-    #  (github.event_name != 'pull_request' || github.event.pull_request.merged == true || github.event.action != 'closed')
+    if: |
+      needs.process-labels.outputs.has-backend == 'true' && 
+      (github.event_name != 'pull_request' || github.event.pull_request.merged == true || github.event.action != 'closed') &&
```

actionlint (1.7.4)

context "matrix" is not allowed here. available contexts are "github", "inputs", "needs", "vars". see https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability for more details

(expression)